### PR TITLE
Dynamic geometry updater inheritance

### PR DESCRIPTION
**Pull request by hpinkos**
Use inheritance for `DynamicGeometryUpdater` classes to reduce duplicate code
I think this will be the last of my DataSource refactor/cleanup pull requests.

TODO
- [x] Merge #6246
- [x] Merge #6249

### CODE REVIEW

**Comment by cesium-concierge** [comment]
Signed CLA is on file.

@hpinkos, thanks for the pull request! Maintainers, we have a signed CLA from @hpinkos, so you can review this at any time.

:warning: I noticed that [CHANGES.md](https://github.com/AnalyticalGraphicsInc/cesium/blob/master/CHANGES.md) has not been updated. If this change updates the public API in any way, fixes a bug, or makes any non-trivial update, please add a bullet point to `CHANGES.md` and comment on this pull request so we know it was updated. For more info, see the [Pull Request Guidelines]( https://github.com/AnalyticalGraphicsInc/cesium/blob/master/CONTRIBUTING.md#pull-request-guidelines).

---

__I am a bot who helps you make Cesium awesome! [Contributions to my configuration are welcome.](https://github.com/AnalyticalGraphicsInc/cesium/tree/master/.concierge)__

:earth_africa: :earth_americas: :earth_asia:

**Comment by mramato** [comment]
Please merge in master.

**Comment by hpinkos** [comment]
@mramato ready

**Comment by mramato** [comment]
This example looks broken: http://localhost:8080/Apps/Sandcastle/index.html?src=Rectangle.html&label=Geometries  I actually think it was one of your previous PRs that broke it (master is broken too but worked 7 commits ago).  Might as well fix it now.

File: Source/DataSources/EllipsoidGeometryUpdater.js [hunk 1]
```diff
@@ -228,7 +224,13 @@ define([
         this._material = {};
     }
 
-    DynamicGeometryUpdater.prototype.update = function(time) {
+
```
Comment (mramato) [inline Source/DataSources/EllipsoidGeometryUpdater.js]: Whitespace

**Comment by hpinkos** [comment]
@mramato just fixed the rectangle issue

**Comment by hpinkos** [comment]
This should be ready now

**Comment by mramato** [comment]
Thanks!

**Pull request merged**